torch/_inductor/ir.py

```diff
@@ -1827,6 +1827,52 @@ def is_stride_order_storage_and_layout(x, stride_order):
         return False
 
 
+def is_pointwise_contiguous_or_transposed_after_perm(x):
+    if (
+        isinstance(x, PermuteView)
+        and isinstance(x.data, StorageBox)
+        and isinstance(x.data.data, Pointwise)
+    ):
+        all_reads = x.data.data.get_reads()
+
+        def is_contiguous_or_transposed_after_perm(read):
+            def get_perm_dims(read):
+                # process perm dim if its length does not equal to reader number
+                perm_dims = x.dims
+                while len(perm_dims) != len(read.size):
+                    if perm_dims[0] != 0:
+                        return []
+                    perm_dims = [d - 1 for d in perm_dims[1:]]
```

Comment:
If my understanding is right here, for example:
```
read.size = [1, 2, 3]
permute_dim = [0, 3, 1, 2]
```
We want to change `permute_dim` to [2, 0, 1].
It seems we have the assumption the missing index of reads is at the first several dimensions and we hardcode the missing reads number as 1?

Reply:
Thank you. Updated as discussed offline.